ui/src/containers/StartTimeProvider.js

```diff
@@ -30,21 +30,22 @@ export const useStartingTimeStamp = (): {
 };
 
 const StartTimeProvider = ({ children }: { children: Node }) => {
-  const { duration } = useMetricsTimeSpan();
+  const { duration, frequency } = useMetricsTimeSpan();
 
-  const [currentTime, setCurrentTime] = useState(new Date());
+  const [currentTime, setCurrentTime] = useState(new Date().getTime());
 
   const [startingTimeISO, setStartingTimeISO] = useState(
     new Date((currentTime / 1000 - duration) * 1000).toISOString(),
   );
 
   const updateCurrentTime = useCallback(() => {
-    const newCurrentDate = new Date();
-    setCurrentTime(newCurrentDate);
+    const newCurrentDate = new Date().getTime();
+    const newCurrentTime = newCurrentDate - (newCurrentDate % (frequency * 1000));
```

Comment:
I think here it worth writiing down why we did this change (what problem we were trying to solve)